tests/ResourceSyncingUsingPolymorphicTest.php

```diff
@@ -0,0 +1,333 @@
+        'global_id' => 'acme',
+        'name' => 'ArchTech',
+        'email' => 'archtech@localhost',
+    ]);
+
+    $tenant2->run(function () {
+        expect(TenantCompanyUsingPolymorphic::all())->toHaveCount(0);
+    });
+
+    $centralCompany->tenants()->attach('t2');
+
+    // Assert Company resource is synced
+    $tenant2->run(function () use ($centralCompany) {
+        $tenantCompany = TenantCompanyUsingPolymorphic::first()->only(['name', 'email']);
+        $centralCompany = $centralCompany->only(['name', 'email']);
+
+        expect($tenantCompany)->toBe($centralCompany);
+    });
+});
+
+test('resource syncing works using a single pivot table for multiple models when syncing from tenant to central', function () {
+    $tenant1 = ResourceTenantUsingPolymorphic::create(['id' => 't1']);
+    migrateUsersTableForTenants();
+
+    tenancy()->initialize($tenant1);
+
+    $tenantUser = TenantUserUsingPolymorphic::create([
+        'name' => 'John Doe',
+        'email' => 'john@localhost',
+        'password' => 'password',
+        'role' => 'commenter',
+    ]);
+
+    tenancy()->end();
+
+    // Assert User resource is synced
+    $centralUser = CentralUserUsingPolymorphic::first()->only(['name', 'email', 'password', 'role']);
+    $tenantUser = $tenantUser->only(['name', 'email', 'password', 'role']);
+    expect($tenantUser)->toBe($centralUser);
+
+    $tenant2 = ResourceTenantUsingPolymorphic::create(['id' => 't2']);
+    migrateCompaniesTableForTenants();
+
+    tenancy()->initialize($tenant2);
+
+    $tenantCompany = TenantCompanyUsingPolymorphic::create([
+        'global_id' => 'acme',
+        'name' => 'tenant comp',
+        'email' => 'company@localhost',
+    ]);
+
+    tenancy()->end();
+
+    // Assert Company resource is synced
+    $centralCompany = CentralCompanyUsingPolymorphic::first()->only(['name', 'email']);
+    $tenantCompany = $tenantCompany->only(['name', 'email']);
+    expect($tenantCompany)->toBe($centralCompany);
+});
+
+function migrateCompaniesTableForTenants(): void
+{
+    pest()->artisan('tenants:migrate', [
+        '--path' => __DIR__ . '/Etc/synced_resource_migrations/companies',
+        '--realpath' => true,
+    ])->assertExitCode(0);
+}
+
+// Tenant model used for resource syncing setup
+class ResourceTenantUsingPolymorphic extends Tenant
+{
+    public function users(): MorphToMany
+    {
+        return $this->morphedByMany(CentralUserUsingPolymorphic::class, 'tenant_resources', 'tenant_resources', 'tenant_id', 'resource_global_id', 'id', 'global_id')
+            ->using(TenantPivot::class);
+    }
+
+    public function companies(): MorphToMany
+    {
+        return $this->morphedByMany(CentralCompanyUsingPolymorphic::class, 'tenant_resources', 'tenant_resources', 'tenant_id', 'resource_global_id', 'id', 'global_id')
+            ->using(TenantPivot::class);
+    }
+}
+
+class CentralUserUsingPolymorphic extends Model implements SyncMaster
```

Comment:
Can you also check that you can access the `tenants()` relationship from these models?

Reply:
Assertions added.